include/igl/opengl/ViewerData.cpp

```diff
@@ -265,24 +263,25 @@ IGL_INLINE void igl::opengl::ViewerData::set_texture(
 }
 
 IGL_INLINE void igl::opengl::ViewerData::set_data(
-  const double caxis_min,
-  const double caxis_max,
-  const Eigen::VectorXd & D)
+  const Eigen::VectorXd & D,
+  double caxis_min,
+  double caxis_max,
+  igl::ColorMapType cmap)
 {
   if(!show_texture)
   {
     Eigen::MatrixXd CM;
-    igl::parula(Eigen::VectorXd::LinSpaced(21,0,1).eval(),false,CM);
+    igl::colormap(cmap,Eigen::VectorXd::LinSpaced(256,0,1).eval(),0,1,CM);
```

Comment:
this `256` should be reverted `21`.

Intervals are (by default) better for visualizing scalar fields than an effectively continuous map.

Reply:
Ok I see. I've added an argument to set the number of steps/interval then, with default = 21.